### DISCUSSION TITLE
Modifying syntax highlighting

### DISCUSSION THREAD

**Discussion by advice2020** [geany/geany#3996] [Q&A]
I was wondering if anyone would be willing to be able to answer some questions about how to modify syntax highlighting.

I would like to try modifying a couple different languages syntax but just for an example for this post, I would like to focus on XML syntax.

1.) CURRENT LINE HIGHLIGHT - FOREGROUND COLOR
For example this is the line highlight
`current_line=#4D4D4D;#FFFFFF;true;false`
The 2nd color is for the line background highlight, but in this case does the 1st (foreground) color do anything?  I thought it might apply to text but it does not appear to.

2.) CURRENT LINE HIGHLIGHT - TRANSPARENT
Does anyone know if there are any transparent color codes available.  I have seen that you can add two letters/digits in front of color codes to change the transparency but have not been able to get this to work.  I have also seen "translucency" in manual `translucency=256;256;false;false` but do not understand how this is applied.  If this is possible can someone show how this can be achieved with the following example
`current_line=red;cyan;true;false`
I tried putting this line below this line
`translucency=100;100;false;false`
but it did not appear to do anything.
It says in the manual that it is for the "current line".

Reason for this is I figured out how to add a background color to tags, just had to add a second color code to the "tag=" for example "tag=red;green" but when you use a "current_line=" background highlight it covers the tag colors completely.  Not important but was just curious if could make the background highlight of the line be transparent to allow the tag background colors to be slightly seen, just wanted to see how this would look.

Thank you for any information shared.

**Comment by elextr** [comment]
Lexing is provided by the Lexilla project which is C++.  Anything that needs different lexing will have to be accepted in that project.

Geany does not support more than one font set by the `View` menu.

See how example colour schemes work in Geany-themes repository.

**Comment by advice2020** [comment]
Thank You
I knew about the Geany themes already, I knew they changed the syntax highlighting but the naming of this still throws me off even though I know this.  I still keep thinking when I hear Geany themes that they are themes to change Geany GUI, but they are for the typing pane/syntax highlighting.
I was able to figure out the answers to many of my questions, I edited my original post with some things I could not figure out if you or anyone else is willing to provide any help regarding this topic.

Thanks again

**Comment by elextr** [comment]
They are called "colour schemes" (see the menu) which is not confused with GTK UI themes, but a historical mistake named the repository "themes" instead of colour schemes and unfortunately nobody has renamed it since (and I don't even know it it can be renamed).
